### PR TITLE
Goodbye, #talent-pool

### LINKED ISSUE #1422: Remove talent pool watch channel
The nominations watch channel should be removed, it isn't used so is not necessary to have any more. All the other nomination logic should stay, we just don't need to relay nominated users' messages to a channel.

### PR DESCRIPTION
**Pull request by ks129**
Closes #1422 

- Migrated TalentPool cog to not use WatchChannel cog as a parent class.
- Migrated Reviewer class to apply changes from TalentPool cog migration.
- Renamed some commands (left aliases) so they are not referencing to watching anymore.
- Implemented cog's cache.
- Removed #talent-pool configuration.

### CODE REVIEW

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:115]: This exact code exists in the watch channels code. Can you refactor this into a utility function that both can use? Not sure what the best location for it would be.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:35]: This entire function is duplicated in watch channels. Can you also refactor this into a shared function?

**Review by MarkKoz — CHANGES_REQUESTED** [review]

**Comment by Xithrius** [comment]
In my test, this PR works. Needs more testing and more eyes.

File: bot/exts/recruitment/talentpool/_review.py
Comment (ChrisLovering) [inline bot/exts/recruitment/talentpool/_review.py:68]: Does this and the lines below reintroduce the bug fixed in https://github.com/python-discord/bot/pull/1569

Comment (Xithrius) [reply bot/exts/recruitment/talentpool/_review.py:68]: I haven't experienced them myself, but I will change to `.get` just in case, after testing.

Comment (Xithrius) [reply bot/exts/recruitment/talentpool/_review.py]: Resolved in d1a2d9965fadc5292685d3ca78a75fd16f07af26

**Comment by Xithrius** [comment]
@MarkKoz I think the current duplicated functions should remain where they are for now. If we do decide to move these functions somewhere else, it should be within a different PR focused on refactoring.

The functions themselves in my own attempt of refactoring ended up taking a bit too many variables as arguments. That may have just been me though, and it was quite a while since said last attempt. Although at this point in time it seems to be still the same case.

**Comment by MarkKoz** [comment]
> 
> 
> @MarkKoz I think the current duplicated functions should remain where they are for now. If we do decide to move these functions somewhere else, it should be within a different PR focused on refactoring.
> 
> The functions themselves in my own attempt of refactoring ended up taking a bit too many variables as arguments. That may have just been me though, and it was quite a while since said last attempt. Although at this point in time it seems to be still the same case.

Thanks for looking into that. It's unfortunate that a nice solution cannot be found, but it ultimately is not that much code that is duplicated. It shouldn't hold up the PR any longer. I will get to testing it right now.

File: bot/exts/recruitment/talentpool/_review.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_review.py:106]: The variable names this comment references should be updated.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:45]: These users aren't being "watched" anymore.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:153]: The name should be "add". "watch" can be added as an alias if desired.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:277]: This function should be renamed.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (MarkKoz) [inline bot/exts/recruitment/talentpool/_cog.py:26]: Update the docstring; it's inaccurate now.

**Review by MarkKoz — CHANGES_REQUESTED** [review]
The endpoint URL needs to be updated here

https://github.com/python-discord/bot/blob/d1a2d9965fadc5292685d3ca78a75fd16f07af26/bot/exts/recruitment/talentpool/_review.py#L95-L97

This command's primary name (which affects the help command) shouldn't have "watch" in it. Furthermore, the function should be renamed.

https://github.com/python-discord/bot/blob/d1a2d9965fadc5292685d3ca78a75fd16f07af26/bot/exts/recruitment/talentpool/_cog.py#L143-L145

Mentions of "watch" should be removed from here, including the function name.

https://github.com/python-discord/bot/blob/d1a2d9965fadc5292685d3ca78a75fd16f07af26/bot/exts/recruitment/talentpool/_cog.py#L174-L178

This should not say "unwatch"

https://github.com/python-discord/bot/blob/d1a2d9965fadc5292685d3ca78a75fd16f07af26/bot/exts/recruitment/talentpool/_cog.py#L477

**Comment by wookie184** [comment]
@Xithrius do you have a status on when you might be able to respond to Mark's comments/finish this up? It seems pretty close and would be nice to get it merged fairly soon. Thanks!

**Comment by Xithrius** [comment]
> @Xithrius do you have a status on when you might be able to respond to Mark's comments/finish this up? It seems pretty close and would be nice to get it merged fairly soon. Thanks!

I'm still in the process of slowly resolving all of this. At the same time, I will also be needing to resolve the current file conflict. I also hope to merge this soon.

**Comment by mbaruh** [comment]
> 
> 
> @MarkKoz I think the current duplicated functions should remain where they are for now. If we do decide to move these functions somewhere else, it should be within a different PR focused on refactoring.
> 
> The functions themselves in my own attempt of refactoring ended up taking a bit too many variables as arguments. That may have just been me though, and it was quite a while since said last attempt. Although at this point in time it seems to be still the same case.

I feel like some of the mess due to me rushing the auto-reviewer before the PR could be merged.

One possible solution is to have a base class "Register" (or some better name) that will have all the functionality related to adding, removing, and displaying a list of the added users. The talentpool and watch cogs would then inherit from it and use it to implement the `watch`, `unwatch`, and `list commands`.

**Comment by Xithrius** [comment]
Me and @ChrisLovering are now collaborating on this PR to get it finished faster.

**Review by MarkKoz — APPROVED** [review]
I hope I tested everything. It's a fairly involved system and I don't understand it all since I don't regularly use it.

**Comment by MarkKoz** [comment]
Oh, there's this comment from @mbaruh but I guess it is out of scope for the PR? I'm not sure what exactly the suggested change is anyway — I'm not too familiar with the whole system anymore.

https://github.com/python-discord/bot/blob/9ef8306c2bc8ab0285da62ba2c7c42e003f015e2/bot/exts/recruitment/talentpool/_review.py#L56-L65

**Comment by wookie184** [comment]
Not suggesting it should be changed in this PR, but do we actually need a form of cache at all for this? All the cases it is used it seems to be refreshed before use (unless i'm missing something), which seems to sort of defeats the point of caching it.

There also seem to be quite a few validation requests before making the site request (often meaning extra api requests are required). As the site already does validation it would probably be much easier just to handle those failures (although i'm not sure how granular the responses are atm, might require addition of status codes returned from the site side).

Anyway, neither of those things are really relevant to this PR, so i'll try and get round to testing this, code looks fine from a quick skim.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (wookie184) [inline bot/exts/recruitment/talentpool/_cog.py:544]: Would `Unnomination reason:` rather than `Unnominate reason:` be slightly better english? I'm not actually sure

File: bot/exts/recruitment/talentpool/_cog.py
Comment (wookie184) [inline bot/exts/recruitment/talentpool/_cog.py:465]: We'd probably want to pop the user from the cache here, I guess, although in practice it probably wouldn't affect much.

**Review by wookie184 — CHANGES_REQUESTED** [review]
Looked through the code and tested it out- just two small comments, other than that all looks fine.

File: bot/exts/recruitment/talentpool/_cog.py
Comment (ChrisLovering) [reply bot/exts/recruitment/talentpool/_cog.py:544]: Yes, that would be correct. Will update later

File: bot/exts/recruitment/talentpool/_cog.py
Comment (ChrisLovering) [reply bot/exts/recruitment/talentpool/_cog.py:465]: Yea agreed, pushed to https://github.com/python-discord/bot/pull/1487/commits/a06367e1691d8ef39e2cf8f8e0d15e5922d29d01

**Review by wookie184 — APPROVED** [review]
LGTM

**Pull request merged**
